toy_robot.rb

```diff
@@ -7,34 +7,47 @@ def initialize
     @x = 0
     @y = 0
     @facing = :north
+    @movements = []
   end
 
   def place(x,y,facing)
     raise InvalidPositionError.new("Invalid Position, #{x}, #{y}, #{facing}") if !(x.between?(0,4)) || !(y.between?(0,4))
     @x = x
     @y = y
     @facing = facing
+    @movements.replace(["PLACE #{@x}, #{@y}, #{@facing.to_s.upcase}"])
   end
 
   def move
```

Comment:
The validation code you have in the place method looks good. You could move this into a validate(x,y,f) function. Then you could call this function from move. You would first calculate the new x and y in temp variables then validate them. If validation successful you can then update the class level properties and the movement history.